lib/auth/methods.go

```diff
@@ -406,10 +406,14 @@ func (a *Server) authenticateUserInternal(
 
 	// When using password only make sure that auth preference does not require
 	// second factor, otherwise users could bypass it.
-	switch authPreference.GetSecondFactor() {
-	case constants.SecondFactorOff:
-		// No 2FA required, check password only.
-	case constants.SecondFactorOptional:
+	switch {
+	case authPreference.IsSecondFactorEnforced():
```

Comment:
I'm surprised to see this code fragment at all. Didn't we enforce MFAs globally a while ago?

Reply:
Yeah exactly:

```
// IsSecondFactorEnforced checks if second factor is enabled.
func (c *AuthPreferenceV2) IsSecondFactorEnabled() bool {
	// TODO(Joerger): outside of tests, second factor should always be enabled.
	// All calls should be removed and the old off/optional second factors removed.
	return len(c.GetSecondFactors()) > 0
}
```